pkg/util/kubernetes/apiserver/util.go

```diff
@@ -22,15 +23,18 @@ var syncTimeout = config.Datadog.GetDuration("cache_sync_timeout") * time.Second
 
 // SyncInformers should be called after the instanciation of new informers.
 // It's blocking until the informers are synced or the timeout exceeded.
-func SyncInformers(informers map[string]cache.SharedInformer) error {
+func SyncInformers(informers map[InformerName]cache.SharedInformer) error {
 	var g errgroup.Group
-	for name, inf := range informers {
+	for name := range informers {
+		name := name // https://golang.org/doc/faq#closures_and_goroutines
 		g.Go(func() error {
-			ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(syncTimeout))
+			ctx, cancel := context.WithTimeout(context.Background(), syncTimeout)
 			defer cancel()
-			if !cache.WaitForCacheSync(ctx.Done(), inf.HasSynced) {
-				return log.Errorf("cache sync timed out for the %s informer", name)
+			log.Debugf("Sync started for informer %s", name)
+			if !cache.WaitForCacheSync(ctx.Done(), informers[name].HasSynced) {
+				return fmt.Errorf("cache sync timed out for informer %s", name)
 			}
+			log.Debugf("Sync done for informer %s", name)
```

Comment:
The numbeer of debug logs can be decreased here.
For example:
- store the start time: `start := time.Now()` line 33
- create a log only when the sync is done, and do a time diff know the duration.
- maybe add the same log in case of timeout

WDYT?

Reply:
sounds good, will do